### PR TITLE
Limit in Production to either a Station, Clinic, or Facility

### PR DESCRIPTION
**Pull request by brianseek**
## Summary
Adds ability to filter travel pay by station, facility, and clinic.

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#52868

## Testing done

Added unit test

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature
- [ ]  [Accessibility foundational testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### CODE REVIEW

**Review by acrollet — CHANGES_REQUESTED** [review]
So the logic and tests look great to me, the one thing that stuck out to me is that I had assumed we would match on facility ID and clinic IEN rather than names. As long as we're doing string matching anyway I don't think the logic should need the change, imo the one thing to do would be to switch to `clinicIen` rather than clinic name. Here's some staging sample data, fwiw on production I see e.g. `"facility": "564GC"`:

```
            {
                "clinicName": "STG/CARDIO/1",
                "appointmentIen": "37862",
                "stationNo": "530",
                "clinicIen": "451",
                "checkedInTime": "",
                "kind": "clinic",
                "eligibility": "ELIGIBLE",
                "clinicPhoneNumber": "555-555-5555",
                "clinicLocation": "SECOND FLOOR ROOM 2",
                "clinicStopCodeName": "CARDIOLOGY",
                "doctorName": "",
                "checkInSteps": [],
                "clinicFriendlyName": "HEART CLINIC 1",
                "startTime": "2023-02-21T10:30:00.000-05:00",
                "clinicCreditStopCodeName": "",
                "facility": "VISTA-API STAGING",
                "checkInWindowEnd": "2023-02-21T10:45:00.000-05:00",
                "checkInWindowStart": "2023-02-21T09:45:00.000-05:00",
                "status": "NO ACTION TAKEN/TODAY"
            }
```

Let me know what you think.

**Comment by brianseek** [comment]
> So the logic and tests look great to me, the one thing that stuck out to me is that I had assumed we would match on facility ID and clinic IEN rather than names. As long as we're doing string matching anyway I don't think the logic should need the change, imo the one thing to do would be to switch to `clinicIen` rather than clinic name. Here's some staging sample data, fwiw on production I see e.g. `"facility": "564GC"`:
> 
> ```
>             {
>                 "clinicName": "STG/CARDIO/1",
>                 "appointmentIen": "37862",
>                 "stationNo": "530",
>                 "clinicIen": "451",
>                 "checkedInTime": "",
>                 "kind": "clinic",
>                 "eligibility": "ELIGIBLE",
>                 "clinicPhoneNumber": "555-555-5555",
>                 "clinicLocation": "SECOND FLOOR ROOM 2",
>                 "clinicStopCodeName": "CARDIOLOGY",
>                 "doctorName": "",
>                 "checkInSteps": [],
>                 "clinicFriendlyName": "HEART CLINIC 1",
>                 "startTime": "2023-02-21T10:30:00.000-05:00",
>                 "clinicCreditStopCodeName": "",
>                 "facility": "VISTA-API STAGING",
>                 "checkInWindowEnd": "2023-02-21T10:45:00.000-05:00",
>                 "checkInWindowStart": "2023-02-21T09:45:00.000-05:00",
>                 "status": "NO ACTION TAKEN/TODAY"
>             }
> ```
> 
> Let me know what you think.

I agree, I must of just missed clinicIen when I started. I was looking at staging data and I didn't see an unique ID for clinic or facility, but it looks like your stage data does have the clinicIen, so I will update with that.

File: src/applications/check-in/utils/appConstants/appConstants.unit.spec.js [hunk 1]
```diff
@@ -48,4 +48,35 @@ describe('check in utils', () => {
...
+      const appointment = {
+        stationNo: '0001',
+        clinicIen: '0001',
+      };
+      expect(isInAllowList(appointment)).to.be.true;
+    });
+    it('should return true only station listed', () => {
+      const appointment = {
+        stationNo: '500',
+        clinicIen: null,
```
Comment (acrollet) [inline src/applications/check-in/utils/appConstants/appConstants.unit.spec.js:77]: Shouldn't `clinicIen` have a non-null value here? It should always have one in LoROTA so I think that's how we would want to validate it.

Comment (brianseek) [reply src/applications/check-in/utils/appConstants/appConstants.unit.spec.js]: Yeah that makes sense. I think my line of thought was something where it clearly wasn't being evaluated. Update pushed!

**Review by acrollet — COMMENTED** [review]
:octocat: looks great!

**Review by acrollet — APPROVED** [review]

**Pull request merged**